File: 프로그래머스/1/388351. 유연근무제/유연근무제.go
func solution(schedules []int, timelogs [][]int, startday int) int {
	after10min := []int{}
	result := len(schedules)

	for _, schedule := range schedules {
		hour := schedule / 100
		min := schedule%100 + 10
		if min >= 60 {
			hour++
			min -= 60
		}
		after10min = append(after10min, hour*100+min)
	}

	for idx, times := range timelogs {
		standard := after10min[idx]
		for dayIdx, time := range times {
			if standard < time && (dayIdx+startday)%7 != 0 && (dayIdx+startday)%7 != 6 {
				result--
				break
			}
		}
	}

	return result
}

File: 프로그래머스/1/388351. 유연근무제/유연근무제.py
def solution(schedules, timelogs, startday):
    result = 0
    for idx, schedule in enumerate(schedules):
        schedules[idx] = changeMin(schedule) + 10


    for pIdx,timelog in enumerate(timelogs):
        check = True
        for dayIdx, log in enumerate(timelog):
            if ((startday+dayIdx) % 7 != 0 and (startday+dayIdx) % 7 != 6) and schedules[pIdx] < changeMin(log):
                check = False
                break
        if check:
            result += 1
    
    return result



def changeMin(time):
    hour = time // 100
    min = (60 * hour) + (time % 100)

    return min